lib/source/membership.rb

```diff
@@ -29,6 +29,9 @@ def merged_with(csv)
 
         pr = reconciler.reconciliation_data rescue abort($!.to_s)
         pr.each { |r| id_map[r[:id]] = r[:uuid] }
+      else
```

Comment:
This `else` bugs me a little (as always). It would be good to split this out into a series of different strategies, each with much simpler logic, but that seems like too big a change for right now. I don't think this makes the existing code significantly more complex (if anything there's now a nice symmetry between lines 31 and 34) where previously there was the dangling question of "but what if there are no `merge_instructions`), so I think we can live with it for a while whilst the whole build process is in flux anyway. (This whole thing is going to need to be completely changed once we start adding `Person`s first, and then building up their `Membership`s afterwards, rather than vice versa)